src/Supply/Application/Dto/UserContext.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright (c) 2018 Adshares sp. z o.o.
+ *
+ * This file is part of AdServer
+ *
+ * AdServer is free software: you can redistribute and/or modify it
+ * under the terms of the GNU General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * AdServer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with AdServer. If not, see <https://www.gnu.org/licenses/>
+ */
+
+declare(strict_types = 1);
+
+namespace Adshares\Supply\Application\Dto;
+
+use Adshares\Adserver\Http\Utils;
+use Generator;
+use InvalidArgumentException;
+use function iterator_to_array;
+
+final class UserContext
+{
+    /** @var array */
+    private $keywords;
+
+    /** @var float */
+    private $humanScore;
+
+    /** @var string */
+    private $userId;
+
+    public function __construct(array $keywords, float $humanScore, string $userId)
+    {
+        $this->keywords = $keywords;
+        $this->humanScore = $humanScore;
+        $this->userId = $userId;
+//        $this->failIfInvalid();
```

Comment:
Is this comment intended?

Reply:
Yes, because the uid has different format for now